src/main/java/com/exasol/adapter/commontests/scalarfunction/TestSetup.java

```diff
@@ -0,0 +1,48 @@
+package com.exasol.adapter.commontests.scalarfunction;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Set;
+
+import com.exasol.adapter.commontests.scalarfunction.virtualschematestsetup.VirtualSchemaTestSetupProvider;
+import com.exasol.adapter.metadata.DataType;
+
+/**
+ * Interface for a dialect specific test setup.
+ */
+public interface TestSetup {
+
+    /**
+     * Returns a set of scalar functions that should not be tested for this dialect.
+     *
+     * @return set of functions
+     */
+    public Set<String> getDialectSpecificExcludes();
+
+    /**
+     * Get a dialect specific test setup provider.
+     *
+     * @return dialect specific {@link VirtualSchemaTestSetupProvider}
+     */
+    public VirtualSchemaTestSetupProvider getVirtualSchemaTestSetupProvider();
```

Comment:
looks like this method is only used for creating VirtualSchemaTestSetUps. perhaps would make sense to make `TestSetup` extend `VirtualSchemaTestSetupProvider` and remove this method

Reply:
I decided here for composition because it allows us to provide abstract implementations at different steps without building a huge stack of  abstract classes